src/js/legacy/mega-menu.js

```diff
@@ -30,6 +38,42 @@ class MegaMenu {
     }
   }
 
+  showMegaMenu() {
+    this.openControl.setAttribute('hidden', 'hidden');
+    this.menu.removeAttribute('hidden');
+    this.closeControl.removeAttribute('hidden');
+    if (!isWideScreen()) {
+      this.firstMenuElement.focus();
+    }
+  }
+
+  hideMegaMenu() {
+    this.closeControl.setAttribute('hidden', 'hidden');
+    this.menu.setAttribute('hidden', 'hidden');
+    this.openControl.removeAttribute('hidden');
+    this.menu.classList.remove('vetnav--submenu-expanded');
+  }
+
+  toggleSmallMegaMenu(e) {
+    if (!isWideScreen() && isEscape(e)) {
+      this.hideMegaMenu();
+      this.openControl.focus();
+    }
+  }
+
+  enterSmallMegaMenu(e) {
+    if (!isWideScreen() && isTab(e) && !e.shiftKey) {
```

Comment:
Does this work on a screenreader? Do they fire key events to change focus?

Reply:
Focus changes as expected using a screenreader, but there are a few discrepancies with Safari that need to be resolved as well (the final menu item doesn't get reached, and is skipped in the tab order), good catch.

Reply:
It looks like the cross-browser issues encountered may be beyond the scope of these changes, more information available [here](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Using_native_keyboard_accessibility).